File: Lunara_backend/server/src/models/HelpArticle.ts
import { Model, DataTypes, Optional } from 'sequelize';
import sequelize from '../config/database';

export interface HelpArticleAttributes {
    id: string;
    title: string;
    content: string;
    category: string;
    isPublished: boolean;
    displayOrder: number;
    createdAt?: Date;
    updatedAt?: Date;
}

export interface HelpArticleCreationAttributes
    extends Optional<HelpArticleAttributes, 'id' | 'isPublished' | 'displayOrder' | 'createdAt' | 'updatedAt'> {}

class HelpArticle
    extends Model<HelpArticleAttributes, HelpArticleCreationAttributes>
    implements HelpArticleAttributes {
    public id!: string;
    public title!: string;
    public content!: string;
    public category!: string;
    public isPublished!: boolean;
    public displayOrder!: number;
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;
}

HelpArticle.init(
    {
        id: {
            type: DataTypes.UUID,
            defaultValue: DataTypes.UUIDV4,
            primaryKey: true,
        },
        title: {
            type: DataTypes.STRING(255),
            allowNull: false,
            validate: {
                notEmpty: { msg: 'Title is required' },
                len: { args: [2, 255], msg: 'Title must be between 2 and 255 characters' },
            },
        },
        content: {
            type: DataTypes.TEXT,
            allowNull: false,
            validate: {
                notEmpty: { msg: 'Content is required' },
            },
        },
        category: {
            type: DataTypes.STRING(100),
            allowNull: false,
            validate: {
                notEmpty: { msg: 'Category is required' },
            },
        },
        isPublished: {
            type: DataTypes.BOOLEAN,
            defaultValue: false,
            field: 'is_published',
        },
        displayOrder: {
            type: DataTypes.INTEGER,
            defaultValue: 0,
            field: 'display_order',
        },
    },
    {
        sequelize,
        tableName: 'help_articles',
        underscored: true,
        timestamps: true,
        indexes: [
            { fields: ['category'] },
            { fields: ['is_published'] },
            { fields: ['display_order'] },
        ],
    }
);

export default HelpArticle;

File: lib/models/help_article.dart
class HelpArticle {
  final String id;
  final String title;
  final String content;
  final String category;
  final bool isPublished;
  final int displayOrder;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  HelpArticle({
    required this.id,
    required this.title,
    required this.content,
    required this.category,
    required this.isPublished,
    required this.displayOrder,
    this.createdAt,
    this.updatedAt,
  });

  factory HelpArticle.fromJson(Map<String, dynamic> json) {
    return HelpArticle(
      id: json['id']?.toString() ?? '',
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      category: json['category'] ?? '',
      isPublished: json['isPublished'] ?? false,
      displayOrder: json['displayOrder'] ?? 0,
      createdAt: json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'content': content,
      'category': category,
      'isPublished': isPublished,
      'displayOrder': displayOrder,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}
